django/conf/urls/__init__.py

```diff
@@ -59,27 +59,108 @@ def include(arg, namespace=None, app_name=None):
             RemovedInDjango20Warning, stacklevel=2
         )
 
-    namespace = namespace or app_name
-
     # Make sure we can iterate through the patterns (without this, some
     # testcases will break).
-    if isinstance(patterns, (list, tuple)):
-        for url_pattern in patterns:
+    if isinstance(urlconf.urlpatterns, (list, tuple)):
+        for pattern in urlconf.urlpatterns:
             # Test if the LocaleRegexURLResolver is used within the include;
             # this should throw an error since this is not allowed!
-            if isinstance(url_pattern, LocaleRegexURLResolver):
-                raise ImproperlyConfigured(
-                    'Using i18n_patterns in an included URLconf is not allowed.')
+            if any(isinstance(constraint, LocalePrefix) for constraint in pattern.constraints):
+                raise ImproperlyConfigured('Using i18n_patterns in an included URLconf is not allowed.')
+
+    return urlconf, namespace
+
 
-    return (urlconf_module, app_name, namespace)
+def url(regex, view, kwargs=None, name=None, decorators=None):
+    from django.urls import RegexPattern, LocalizedRegexPattern
 
+    if isinstance(regex, six.string_types):
+        regex = RegexPattern(regex)
+    elif isinstance(regex, Promise):
+        regex = LocalizedRegexPattern(regex)
+
+    if not isinstance(regex, (list, tuple)):
+        constraints = [regex]
+    else:
+        constraints = regex
 
-def url(regex, view, kwargs=None, name=None):
     if isinstance(view, (list, tuple)):
-        # For include(...) processing.
-        urlconf_module, app_name, namespace = view
-        return RegexURLResolver(regex, urlconf_module, kwargs, app_name=app_name, namespace=namespace)
+        urlconf, namespace = view
+        return URLPattern(constraints, Include(urlconf, namespace, kwargs, decorators))
     elif callable(view):
-        return RegexURLPattern(regex, view, kwargs, name)
+        return URLPattern(constraints, Endpoint(view, name, kwargs=kwargs, decorators=decorators))
     else:
         raise TypeError('view must be a callable or a list/tuple in the case of include().')
+
+
+class Endpoint(object):
+    def __init__(self, view, name=None, kwargs=None, decorators=None):
+        self.view = view
+        self.name = name
+        self.kwargs = kwargs or {}
+        self.decorators = list(decorators) if decorators is not None else []
+
+    @cached_property
+    def lookup_str(self):
+        """
+        A string that identifies the view (e.g. 'path.to.view_function' or
+        'path.to.ClassBasedView').
+        """
+        func = self.view
+        if isinstance(func, functools.partial):
+            func = func.func
```

Comment:
regarding what's following : 
what if `func.func` is another partial (etc) ; I think it could be but I could be wrong both in that and also in my question (you could want to use the partial class name, though I don't think so) ?